src/ua_util_internal.h

```diff
@@ -86,6 +87,48 @@ typedef UA_UInt64 u64;
 typedef UA_Int64 i64;
 typedef UA_StatusCode status;
 
+/**
+ * Error checking macros
+ */
+
+#define UA_CHECK(STATUSCODE, EVAL_ON_ERROR)                                              \
+    if((STATUSCODE) != UA_STATUSCODE_GOOD) {                                             \
+        EVAL_ON_ERROR;                                                                   \
+    }
+
+#define UA_CHECK_LOG_INTERNAL(STATUSCODE, EVAL, LOG, LOGGER, CAT, MSG, ...)              \
+    UA_MACRO_EXPAND(UA_CHECK(                                                            \
+        STATUSCODE, LOG(LOGGER, CAT, "" MSG "%s (%s:%d: statuscode: %s)", __VA_ARGS__,   \
+                        __FILE__, __LINE__, UA_StatusCode_name(STATUSCODE));             \
```

Comment:
adding `UA_StatusCode_name(STATUSCODE)` is a huge benefit for error logging. If there is no specific macro for statuscode checking then this would not be possible.

Reply:
👍